app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  def update_author_posts_counter
+    author.update(posts_counter: author.posts.count)
+  end
```

Comment:
- [OPTIONAL] Great job with your code. Below code also look awesome. :point_down: 
```
  def update_post_counter
     author.increment!(:posts_counter)
  end

```

- [OPTIONAL] It is a good practice to use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_post_counter method every time a new post has been saved to database, then make the update_post_counter method private. It would look something similar to this. 

```
  # app/models/post.rb
   after_save :update_post_counter

    private

    def update_post_counter
       author.increment!(:posts_counter)
    end

```